## Summary

### tl;dr

Notes:
* The Redis project is owned by a company (Redis) and was relicensed on 2024-03-20
* Caveat: This is purely a summary of the data that has not been validated from anyone within the community
* Analysis of contributions to one repo: https://github.com/redis/redis

The majority of contributions to Redis have always come from Redis employees; however, this has become even more pronounced after Redis relicensed in May 2024. Before the relicense, there were significant contributions from employees of other companies. Here are a few notable examples of contributions in the year leading up to the relicense from employees of Amazon, Alibaba, Huawei, Tencent, and Ericsson. All of these people have transitioned from Redis to the Valkey fork with madolson, enjoy-binbin, and zuiderkwast as top contributors to Valkey.

|People|Company|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
| lyq2333 | Alibaba | 10 | 349 | 190 |
| soloestoy | Alibaba | 31 | 828 | 364 |
| hwware | Huawei Technologies | 25 | 763 | 224 |
| roshkhatri | Amazon | 10 | 1917 | 732 |
| hpatro | Amazon | 14 | 1437 | 632 |
| madolson | Amazon | 24 | 3310 | 1636 |
| enjoy-binbin | Tencent Cloud | 146 | 3502 | 1115 |
| zuiderkwast | Ericsson Software Technology | 16 | 13057 | 10622 |

### After relicense (2024-03-20 - 2024-08-21)

All of the external contributors from Amazon, Alibaba, Tencent, Huawei, and Ericsson who contributed over 10 commits in the year leading up to the relicense mostly stopped contributing (some made 1-3 commits shortly after the relicense, which likely indicated work already in progress). 

Redis - people with >= 5 commits: 
* People: 5 0.1388888888888889 % of people
* Commits: 112
* Additions: 31664 0.7905327807459929 % of total additions
* Deletions: 9822 0.6975852272727273 % of total deletions

Other - people with >= 5 commits: 
* People: 0 0.0 % of people
* Commits: 0
* Additions: 0 0.0 % of total additions
* Deletions: 0 0.0 % of total deletions

Totals in dataset of people with >=5 commits:
* 0.7905327807459929 % of total additions
* 0.6975852272727273 % of total deletions

### 1 year before relicense (2023-03-20 - 2024-03-20)

Redis - people with >= 5 commits: 
* People: 6 0.06451612903225806 % of people
* Commits: 164
* Additions: 189656 0.8011489882989059 % of total additions
* Deletions: 83122 0.7389279047026402 % of total deletions

Other - people with >= 5 commits: 
* People: 12 0.12903225806451613 % of people
* Commits: 319
* Additions: 28334 0.11968909728382546 % of total additions
* Deletions: 16684 0.1483154058138501 % of total deletions

Totals in dataset of people with >=5 commits:
* 0.9208380855827314 % of total additions
* 0.8872433105164903 % of total deletions

External Contributors with 5+ commits:
|People|Company|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
| CharlesChen888 | Looking for a job | 21 | 1362 | 529 |
| lyq2333 | Alibaba | 10 | 349 | 190 |
| meiravgri | None | 5 | 1272 | 361 |
| soloestoy | Alibaba | 31 | 828 | 364 |
| judeng | None | 10 | 489 | 275 |
| hwware | Huawei Technologies | 25 | 763 | 224 |
| moshekaplan | None | 7 | 48 | 4 |
| roshkhatri | Amazon | 10 | 1917 | 732 |
| hpatro | Amazon | 14 | 1437 | 632 |
| madolson | Amazon | 24 | 3310 | 1636 |
| enjoy-binbin | Tencent Cloud | 146 | 3502 | 1115 |
| zuiderkwast | Ericsson Software Technology | 16 | 13057 | 10622 |

### 2 years before relicense (2022-03-20 - 2024-03-20)

Redis - people with >= 5 commits: 
* People: 12 0.06521739130434782 % of people
* Commits: 458
* Additions: 244746 0.7560399232672579 % of total additions
* Deletions: 94566 0.7139104044933642 % of total deletions

Other - people with >= 5 commits: 
* People: 21 0.11413043478260869 % of people
* Commits: 649
* Additions: 49395 0.15258509642562576 % of total additions
* Deletions: 24089 0.1818559284926998 % of total deletions

Totals in dataset of people with >=5 commits:
* 0.9086250196928837 % of total additions
* 0.8957663329860639 % of total deletions

# After relicense (2024-03-20 - 2024-08-21)

In [7]:
from pprint import pprint
import collections
import pandas as pd
import pickle

# Pickle files generated by this script:
# https://github.com/geekygirldawn/project-api-metrics/blob/main/scripts/commits_people.py

people_pickle_a = '../data-files/redis_people_2024-03-20T00:00:00.000+00:002024-08-21T00:00:00.000+00:00.pkl'

with open(people_pickle_a, 'rb') as f:
    person_dict_a = pickle.load(f)

In [8]:
people = len(person_dict_a)
commits = 0
additions = 0
deletions = 0

for key,value in person_dict_a.items():
    # Normalize company names and use emails to derive Amazon affiliations
    if value['company'] == None:
        for email in value['email']:
            if any(x in email.lower() for x in ['redis.com', 'redislabs.com']):
                person_dict_a[key]['company'] = 'Redis Labs'
            if 'amazon.com' in email:
                person_dict_a[key]['company'] = 'Amazon'
    elif 'redis' in value['company'].lower():
        person_dict_a[key]['company'] = 'Redis Labs'
    elif 'alibaba' in value['company'].lower():
        person_dict_a[key]['company'] = 'Alibaba'
    elif any(x in value['company'].lower() for x in ['aws','amazon']):
        person_dict_a[key]['company'] = 'Amazon'
    
        
    # Get descriptive statistics
    commits = commits + value['commits']
    additions = additions + value['additions']
    deletions = deletions + value['deletions']
    
print("People:", people)
print("Commits:", commits)
print("Additions", additions)
print("Deletions", deletions)

People: 36
Commits: 157
Additions 40054
Deletions 14080


In [10]:
for key,value in person_dict_a.items():
    if (value['commits'] >= 5) and (value['company'] == None):
        print(key,value)

tezc {'deletions': 3227, 'company': None, 'additions': 5119, 'commits': 15, 'name': 'Ozan Tezcan', 'email': ['ozantezcan@gmail.com']}


In [11]:
# Manual Fixes
person_dict_a['YaacovHazan']['company'] = 'Redis Labs' # https://www.linkedin.com/in/yaacov-hazan-b8043a99/
person_dict_a['tezc']['company'] = 'Redis Labs' # confirmed in conversation with Madelyn Olson on 2024-09-18

# Remove bots
try:
    del person_dict_a['dependabot[bot]']
except:
    pass

In [12]:
org_people = 0
org_commits = 0
org_additions = 0
org_deletions = 0

other_people = 0
other_commits = 0
other_additions = 0
other_deletions = 0

for key,value in person_dict_a.items():
    try:
        if value['commits'] >= 5:
            if value['company'] == 'Redis Labs':
                org_people += 1
                org_commits = org_commits + value['commits']
                org_additions = org_additions + value['additions']
                org_deletions = org_deletions + value['deletions']
            else:
                other_people += 1
                other_commits = other_commits + value['commits']
                other_additions = other_additions + value['additions']
                other_deletions = other_deletions + value['deletions']
                print(key,value)
            i+=1
    except:
        pass

print("\nRedis - people with >= 5 commits:", "\n* People:", org_people, org_people/people, "% of people")
print("* Commits:", org_commits)
print("* Additions:", org_additions, org_additions/additions, "% of total additions")
print("* Deletions:", org_deletions, org_deletions/deletions, "% of total deletions")

print("\nOther - people with >= 5 commits:", "\n* People:", other_people, other_people/people, "% of people")
print("* Commits:", other_commits)
print("* Additions:", other_additions, other_additions/additions, "% of total additions")
print("* Deletions:", other_deletions, other_deletions/deletions, "% of total deletions")
      
print("\nTotals in dataset of people with >=5 commits:")
print('*', (other_additions + org_additions)/additions, "% of total additions")
print('*', (other_deletions + org_deletions)/deletions, "% of total deletions")


Redis - people with >= 5 commits: 
* People: 5 0.1388888888888889 % of people
* Commits: 112
* Additions: 31664 0.7905327807459929 % of total additions
* Deletions: 9822 0.6975852272727273 % of total deletions

Other - people with >= 5 commits: 
* People: 0 0.0 % of people
* Commits: 0
* Additions: 0 0.0 % of total additions
* Deletions: 0 0.0 % of total deletions

Totals in dataset of people with >=5 commits:
* 0.7905327807459929 % of total additions
* 0.6975852272727273 % of total deletions


In [13]:
# Make it easy for the print statements to be copied into a Markdown table
print('|People|Company|Commits|Additions|Deletions|')
print('|:---|:---|:---|:---|:---|')
for key,value in person_dict_a.items():
    try:
        if (value['company'] != 'Redis Labs') and (value['additions'] >= 50):
            print('|', key,'|', value['company'], '|', value['commits'],'|', value['additions'],'|', value['deletions'],'|')

    except:
        pass

|People|Company|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
| hpatro | Amazon | 2 | 148 | 23 |
| naglera | Amazon | 2 | 130 | 24 |
| ClaytonNorthey92 | None | 2 | 826 | 12 |
| yveslb | None | 2 | 785 | 31 |
| zuiderkwast | Ericsson Software Technology | 3 | 1608 | 42 |
| enjoy-binbin | Tencent Cloud | 2 | 1058 | 95 |
| bjosv | @Ericsson  | 1 | 413 | 6 |
| vitahlin | None | 2 | 260 | 83 |
| j178 | None | 1 | 198 | 135 |


This is manually copied from above 

|People|Company|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
| hpatro | Amazon | 2 | 148 | 23 |
| naglera | Amazon | 2 | 130 | 24 |
| ClaytonNorthey92 | None | 2 | 826 | 12 |
| yveslb | None | 2 | 785 | 31 |
| zuiderkwast | Ericsson Software Technology | 3 | 1608 | 42 |
| enjoy-binbin | Tencent Cloud | 2 | 1058 | 95 |
| bjosv | @Ericsson  | 1 | 413 | 6 |
| vitahlin | None | 2 | 260 | 83 |
| j178 | None | 1 | 198 | 135 |

# 1 year before relicense (2023-03-20 - 2024-03-20)

In [14]:
from pprint import pprint
import collections
import pandas as pd
import pickle

# Pickle files generated by this script:
# https://github.com/geekygirldawn/project-api-metrics/blob/main/scripts/commits_people.py

people_pickle_1yr = '../data-files/redis_people_2023-03-20T00:00:00.000+00:002024-03-20T00:00:00.000+00:00.pkl'

with open(people_pickle_1yr, 'rb') as f:
    person_dict_1yr = pickle.load(f)
    
len(person_dict_1yr)

93

In [15]:
people = len(person_dict_1yr)
commits = 0
additions = 0
deletions = 0

for key,value in person_dict_1yr.items():
    # Normalize company names and use emails to derive Amazon affiliations
    if value['company'] == None:
        for email in value['email']:
            if any(x in email.lower() for x in ['redis.com', 'redislabs.com']):
                person_dict_1yr[key]['company'] = 'Redis Labs'
            if 'amazon.com' in email:
                person_dict_1yr[key]['company'] = 'Amazon'
    elif 'redis' in value['company'].lower():
        person_dict_1yr[key]['company'] = 'Redis Labs'
    elif 'alibaba' in value['company'].lower():
        person_dict_1yr[key]['company'] = 'Alibaba'
    elif any(x in value['company'].lower() for x in ['aws','amazon']):
        person_dict_1yr[key]['company'] = 'Amazon'
    
        
    # Get descriptive statistics
    commits = commits + value['commits']
    additions = additions + value['additions']
    deletions = deletions + value['deletions']
    
print("People:", people)
print("Commits:", commits)
print("Additions", additions)
print("Deletions", deletions)

People: 93
Commits: 592
Additions 236730
Deletions 112490


In [16]:
for key,value in person_dict_1yr.items():
    if (value['commits'] >= 10) and (value['company'] == None):
        print(key,value)

YaacovHazan {'name': None, 'company': None, 'additions': 159, 'commits': 12, 'email': ['31382944+YaacovHazan@users.noreply.github.com'], 'deletions': 183}
yossigo {'name': 'Yossi Gottlieb', 'company': None, 'additions': 8040, 'commits': 11, 'email': ['yossigo@gmail.com'], 'deletions': 1179}
judeng {'name': None, 'company': None, 'additions': 489, 'commits': 10, 'email': ['abc3844@126.com'], 'deletions': 275}


In [17]:
# Manual Fixes
person_dict_1yr['YaacovHazan']['company'] = 'Redis Labs' # https://www.linkedin.com/in/yaacov-hazan-b8043a99/
person_dict_1yr['yossigo']['company'] = 'Redis Labs'  # https://www.linkedin.com/in/yossi-gottlieb-40842/
person_dict_1yr['tezc']['company'] = 'Redis Labs' # confirmed in conversation with Madelyn Olson on 2024-09-18


# Remove bots
try:
    del person_dict_1yr['dependabot[bot]']
except:
    pass

In [18]:
for key,value in person_dict_1yr.items():
    try:
        if (value['commits'] >= 10) and (value['company'] != 'Redis Labs'):
            print(key,value['company'],value['commits'],value['additions'],value['deletions'])
    except:
        pass

CharlesChen888 Looking for a job 21 1362 529
lyq2333 Alibaba 10 349 190
soloestoy Alibaba 31 828 364
judeng None 10 489 275
hwware Huawei Technologies 25 763 224
roshkhatri Amazon 10 1917 732
hpatro Amazon 14 1437 632
madolson Amazon 24 3310 1636
enjoy-binbin Tencent Cloud 146 3502 1115
zuiderkwast Ericsson Software Technology 16 13057 10622


In [21]:
org_people = 0
org_commits = 0
org_additions = 0
org_deletions = 0

other_people = 0
other_commits = 0
other_additions = 0
other_deletions = 0

for key,value in person_dict_1yr.items():
    try:
        if value['commits'] >= 5:
            if value['company'] == 'Redis Labs':
                org_people += 1
                org_commits = org_commits + value['commits']
                org_additions = org_additions + value['additions']
                org_deletions = org_deletions + value['deletions']
            else:
                other_people += 1
                other_commits = other_commits + value['commits']
                other_additions = other_additions + value['additions']
                other_deletions = other_deletions + value['deletions']
                print(key,value)
            i+=1
    except:
        pass

print("\nRedis - people with >= 5 commits:", "\n* People:", org_people, org_people/people, "% of people")
print("* Commits:", org_commits)
print("* Additions:", org_additions, org_additions/additions, "% of total additions")
print("* Deletions:", org_deletions, org_deletions/deletions, "% of total deletions")

print("\nOther - people with >= 5 commits:", "\n* People:", other_people, other_people/people, "% of people")
print("* Commits:", other_commits)
print("* Additions:", other_additions, other_additions/additions, "% of total additions")
print("* Deletions:", other_deletions, other_deletions/deletions, "% of total deletions")
      
print("\nTotals in dataset of people with >=5 commits:")
print('*', (other_additions + org_additions)/additions, "% of total additions")
print('*', (other_deletions + org_deletions)/deletions, "% of total deletions")

CharlesChen888 {'name': 'Chen Tianjie', 'company': 'Looking for a job', 'additions': 1362, 'commits': 21, 'email': ['chentianjie.ctj@alibaba-inc.com', 'TJ_Chen@outlook.com'], 'deletions': 529}
lyq2333 {'name': 'Yanqi Lv', 'company': 'Alibaba', 'additions': 349, 'commits': 10, 'email': ['lvyanqi.lyq@alibaba-inc.com', '50293466+lyq2333@users.noreply.github.com'], 'deletions': 190}
meiravgri {'name': None, 'company': None, 'additions': 1272, 'commits': 5, 'email': ['109056284+meiravgri@users.noreply.github.com'], 'deletions': 361}
soloestoy {'name': 'zhaozhao.zz', 'company': 'Alibaba', 'additions': 828, 'commits': 31, 'email': ['zhaozhao.zz@alibaba-inc.com'], 'deletions': 364}
judeng {'name': None, 'company': None, 'additions': 489, 'commits': 10, 'email': ['abc3844@126.com'], 'deletions': 275}
hwware {'name': 'Wen Hui', 'company': 'Huawei Technologies', 'additions': 763, 'commits': 25, 'email': ['wen.hui.ware@gmail.com'], 'deletions': 224}
moshekaplan {'name': 'Moshe Kaplan', 'company': 

In [25]:
# Make it easy for the print statements to be copied into a Markdown table
print('|People|Company|Commits|Additions|Deletions|')
print('|:---|:---|:---|:---|:---|')
for key,value in person_dict_1yr.items():
    try:
        if (value['company'] != 'Redis Labs') and (value['commits'] >= 5):
            print('|', key,'|', value['company'], '|', value['commits'],'|', value['additions'],'|', value['deletions'],'|')

    except:
        pass

|People|Company|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
| CharlesChen888 | Looking for a job | 21 | 1362 | 529 |
| lyq2333 | Alibaba | 10 | 349 | 190 |
| meiravgri | None | 5 | 1272 | 361 |
| soloestoy | Alibaba | 31 | 828 | 364 |
| judeng | None | 10 | 489 | 275 |
| hwware | Huawei Technologies | 25 | 763 | 224 |
| moshekaplan | None | 7 | 48 | 4 |
| roshkhatri | Amazon | 10 | 1917 | 732 |
| hpatro | Amazon | 14 | 1437 | 632 |
| madolson | Amazon | 24 | 3310 | 1636 |
| enjoy-binbin | Tencent Cloud | 146 | 3502 | 1115 |
| zuiderkwast | Ericsson Software Technology | 16 | 13057 | 10622 |


Manually copied from above

|People|Company|Commits|Additions|Deletions|
|:---|:---|:---|:---|:---|
| CharlesChen888 | Looking for a job | 21 | 1362 | 529 |
| felipou | None | 1 | 138 | 0 |
| lyq2333 | Alibaba | 10 | 349 | 190 |
| meiravgri | None | 5 | 1272 | 361 |
| slavak | None | 3 | 267 | 75 |
| soloestoy | Alibaba | 31 | 828 | 364 |
| judeng | None | 10 | 489 | 275 |
| hwware | Huawei Technologies | 25 | 763 | 224 |
| roshkhatri | Amazon | 10 | 1917 | 732 |
| hpatro | Amazon | 14 | 1437 | 632 |
| madolson | Amazon | 24 | 3310 | 1636 |
| vitarb | None | 2 | 2312 | 1128 |
| panjf2000 | @gnet-io | 3 | 163 | 73 |
| enjoy-binbin | Tencent Cloud | 146 | 3502 | 1115 |
| zuiderkwast | Ericsson Software Technology | 16 | 13057 | 10622 |

# 2 years before relicense (2022-03-20 - 2024-03-20)

In [26]:
from pprint import pprint
import collections
import pandas as pd
import pickle

# Pickle files generated by this script:
# https://github.com/geekygirldawn/project-api-metrics/blob/main/scripts/commits_people.py

people_pickle_2yr = '../data-files/redis_people_2022-03-20T00:00:00.000+00:002024-03-20T00:00:00.000+00:00.pkl'

with open(people_pickle_2yr, 'rb') as f:
    person_dict_2yr = pickle.load(f)
    
len(person_dict_2yr)

184

In [27]:
people = len(person_dict_2yr)
commits = 0
additions = 0
deletions = 0

for key,value in person_dict_2yr.items():
    # Normalize company names and use emails to derive Amazon affiliations
    if value['company'] == None:
        for email in value['email']:
            if any(x in email.lower() for x in ['redis.com', 'redislabs.com']):
                person_dict_2yr[key]['company'] = 'Redis Labs'
            if 'amazon.com' in email:
                person_dict_2yr[key]['company'] = 'Amazon'
    elif 'redis' in value['company'].lower():
        person_dict_2yr[key]['company'] = 'Redis Labs'
    elif 'alibaba' in value['company'].lower():
        person_dict_2yr[key]['company'] = 'Alibaba'
    elif any(x in value['company'].lower() for x in ['aws','amazon']):
        person_dict_2yr[key]['company'] = 'Amazon'
    
        
    # Get descriptive statistics
    commits = commits + value['commits']
    additions = additions + value['additions']
    deletions = deletions + value['deletions']
    
print("People:", people)
print("Commits:", commits)
print("Additions", additions)
print("Deletions", deletions)

People: 184
Commits: 1331
Additions 323721
Deletions 132462


In [28]:
for key,value in person_dict_2yr.items():
    if (value['commits'] >= 10) and (value['company'] == None):
        print(key,value)

YaacovHazan {'additions': 164, 'deletions': 184, 'name': None, 'email': ['31382944+YaacovHazan@users.noreply.github.com'], 'commits': 13, 'company': None}
tezc {'additions': 7944, 'deletions': 569, 'name': 'Ozan Tezcan', 'email': ['ozantezcan@gmail.com'], 'commits': 20, 'company': None}
yossigo {'additions': 10351, 'deletions': 1522, 'name': 'Yossi Gottlieb', 'email': ['yossigo@gmail.com'], 'commits': 25, 'company': None}
dependabot[bot] {'additions': 58, 'deletions': 59, 'name': None, 'email': ['49699333+dependabot[bot]@users.noreply.github.com'], 'commits': 11, 'company': None}
judeng {'additions': 610, 'deletions': 378, 'name': None, 'email': ['abc3844@126.com'], 'commits': 23, 'company': None}


In [29]:
# Manual Fixes
person_dict_2yr['YaacovHazan']['company'] = 'Redis Labs' # https://www.linkedin.com/in/yaacov-hazan-b8043a99/
person_dict_2yr['yossigo']['company'] = 'Redis Labs'  # https://www.linkedin.com/in/yossi-gottlieb-40842/
person_dict_2yr['tezc']['company'] = 'Redis Labs' # confirmed in conversation with Madelyn Olson on 2024-09-18

# Remove bots
try:
    del person_dict_2yr['dependabot[bot]']
except:
    pass

In [30]:
for key,value in person_dict_2yr.items():
    try:
        if (value['commits'] >= 10) and (value['company'] != 'Redis Labs'):
            print(key,value['company'],value['commits'],value['additions'],value['deletions'])
    except:
        pass

CharlesChen888 Looking for a job 25 1701 559
ncghost1 Magic School 10 62 15
pizhenwei ByteDance 22 1576 824
enjoy-binbin Tencent Cloud 263 6749 2457
soloestoy Alibaba 40 1025 511
judeng None 23 610 378
lyq2333 Alibaba 10 349 190
madolson Amazon 67 11318 3320
hpatro Amazon 20 1791 728
hwware Huawei Technologies 57 1205 330
roshkhatri Amazon 11 1964 734
zuiderkwast Ericsson Software Technology 38 15994 11944


In [32]:
org_people = 0
org_commits = 0
org_additions = 0
org_deletions = 0

other_people = 0
other_commits = 0
other_additions = 0
other_deletions = 0

for key,value in person_dict_2yr.items():
    try:
        if value['commits'] >= 5:
            if value['company'] == 'Redis Labs':
                org_people += 1
                org_commits = org_commits + value['commits']
                org_additions = org_additions + value['additions']
                org_deletions = org_deletions + value['deletions']
            else:
                other_people += 1
                other_commits = other_commits + value['commits']
                other_additions = other_additions + value['additions']
                other_deletions = other_deletions + value['deletions']
                print(key,value)
            i+=1
    except:
        pass

print("\nRedis - people with >= 5 commits:", "\n* People:", org_people, org_people/people, "% of people")
print("* Commits:", org_commits)
print("* Additions:", org_additions, org_additions/additions, "% of total additions")
print("* Deletions:", org_deletions, org_deletions/deletions, "% of total deletions")

print("\nOther - people with >= 5 commits:", "\n* People:", other_people, other_people/people, "% of people")
print("* Commits:", other_commits)
print("* Additions:", other_additions, other_additions/additions, "% of total additions")
print("* Deletions:", other_deletions, other_deletions/deletions, "% of total deletions")
      
print("\nTotals in dataset of people with >=5 commits:")
print('*', (other_additions + org_additions)/additions, "% of total additions")
print('*', (other_deletions + org_deletions)/deletions, "% of total deletions")

devnexen {'additions': 139, 'deletions': 17, 'name': 'David CARLIER', 'email': ['devnexen@gmail.com'], 'commits': 7, 'company': None}
CharlesChen888 {'additions': 1701, 'deletions': 559, 'name': 'Chen Tianjie', 'email': ['chentianjie.ctj@alibaba-inc.com', 'TJ_Chen@outlook.com'], 'commits': 25, 'company': 'Looking for a job'}
ncghost1 {'additions': 62, 'deletions': 15, 'name': 'Eriri', 'email': ['275955589@qq.com', 'eriri233@qq.com', '41555481+ncghost1@users.noreply.github.com'], 'commits': 10, 'company': 'Magic School'}
pizhenwei {'additions': 1576, 'deletions': 824, 'name': 'zhenwei pi', 'email': ['pizhenwei@bytedance.com'], 'commits': 22, 'company': 'ByteDance'}
enjoy-binbin {'additions': 6749, 'deletions': 2457, 'name': 'Binbin', 'email': ['binloveplay1314@qq.com'], 'commits': 263, 'company': 'Tencent Cloud'}
soloestoy {'additions': 1025, 'deletions': 511, 'name': 'zhaozhao.zz', 'email': ['zhaozhao.zz@alibaba-inc.com', '276441700@qq.com'], 'commits': 40, 'company': 'Alibaba'}
judeng